bindings/rust/s2n-tls/src/config.rs

```diff
@@ -291,6 +298,57 @@ impl Builder {
         Ok(self)
     }
 
+    /// Corresponds to [s2n_config_add_cert_chain_and_key_to_store].
+    pub fn load_chain(&mut self, chain: CertificateChain<'static>) -> Result<&mut Self, Error> {
+        // Out of an abudance of caution, we hold a reference to the CertificateChain
+        // regardless of whether add_to_store fails or succeeds. We have limited
+        // visibility into the failure modes, so this behavior ensures that _if_
+        // the C library held the reference despite the failure, it would continue
+        // to be valid memory.
+        self.context_mut()
+            .application_owned_certs
+            .push(chain.clone());
```

Comment:
Rather than cloning it would be better to call the add function first (but not check the results yet), push the chain, then check the result.